### PR TITLE
Parse Yul objects

### PR DESCRIPTION
**Pull request by leonardoalt**
This still doesn't support nested Yul objects.

### CODE REVIEW

File: examples/simple_object.yul [hunk 1]
```diff
@@ -0,0 +1,3 @@
+object "runtime" { code { }
```
Comment (chriseth) [inline examples/simple_object.yul:1]: This is very ugly.

File: src/yul_parser.rs [hunk 2]
```diff
@@ -360,19 +422,59 @@ impl Block {
...
             let mut next_identifier = 1u64;
-            Ok(Block::from(pairs.next().unwrap(), &mut next_identifier))
+            Ok(Root::from(pairs.next().unwrap(), &mut next_identifier))
         }
         Err(error) => Err(format!("{}", error)),
     }
 }
 
+pub fn parse_block(source: &str) -> Result<Block, String> {
+    match parse_root(source) {
```
Comment (chriseth) [inline src/yul_parser.rs]: Can't we use `?` here?

File: src/yul_parser.rs [hunk 3]
```diff
@@ -360,19 +422,59 @@ impl Block {
...
         Ok(mut pairs) => {
             let mut next_identifier = 1u64;
-            Ok(Block::from(pairs.next().unwrap(), &mut next_identifier))
+            Ok(Root::from(pairs.next().unwrap(), &mut next_identifier))
         }
         Err(error) => Err(format!("{}", error)),
     }
 }
 
+pub fn parse_block(source: &str) -> Result<Block, String> {
```
Comment (chriseth) [inline src/yul_parser.rs:455]: If we use `yulParser::parse(Rule::block`, we might get better error messages here.

Comment (chriseth) [reply src/yul_parser.rs:455]: Ah, but then we have to add soi eoi and stuff. hm.

File: examples/simple_object.yul (hunk 1 above)
Comment (leonardoalt) [reply examples/simple_object.yul:1]: write a formatter

Comment (chriseth) [reply examples/simple_object.yul:1]: hold my beer

File: src/yul_parser.rs (hunk 2 above)
Comment (leonardoalt) [reply src/yul_parser.rs]: done

File: src/yul_parser.rs (hunk 3 above)
Comment (leonardoalt) [reply src/yul_parser.rs:455]: I can add that back if you want

**Review by chriseth — APPROVED** [review]

**Pull request merged**